docs/pages/kubernetes-access/controls.mdx

```diff
@@ -242,29 +126,81 @@ Label each cluster with key-value pairs describing the cluster:
   </TabItem>
 </Tabs>
 
-Limit access to cluster based on the label:
+### `kubernetes_groups` and `kubernetes_users`
+
+The Teleport Kubernetes Service receives requests from end users, e.g., via
+`kubectl`, and forwards them to a Kubernetes API server. The Kubernetes Service
+uses  [impersonation
+headers](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#user-impersonation)
+to send requests to the API server with one Kubernetes user and zero or more
+Kubernetes groups.
+
+![Impersonation](../../img/k8s/auth.svg)
+
+The `kubernetes_users` and `kubernetes_groups` fields indicate which users and
+groups to allow a user to assume when they send requests to a Kubernetes API
+server:
 
 ```yaml
 kind: role
-version: v5
 metadata:
-  name: group-member
+  name: kube-access
+version: v6
 spec:
   allow:
-    # Allow access to Kubernetes clusters in the production environment
-    kubernetes_labels:
-      'env': 'prod'
-    # Allows access to group "admins" in Kubernetes clusters
-    kubernetes_groups: ["admins"]
+    kubernetes_groups:
+    - developers
+    - viewers
+    kubernetes_users:
+    - myuser
+    # ...
+  deny: {}
 ```
 
-## Impersonation
+The value of `kubernetes_groups` and `kubernetes_users` is a list of names of
+groups and users to enable impersonation for.
+
+#### How Teleport users impersonate Kubernetes users and groups
+
+There are two ways for an end user to specify which user and groups to
+impersonate:
+
+##### Manually
+
+When a user runs `tsh kube login` to authenticate to a Kubernetes cluster, they
+can use the `--as` and `--as-groups` flags to manually specify the user and
+groups to authenticate as. The Teleport Kubernetes Service determines whether
+the user and groups belong to a user's `kubernetes_users` and
+`kubernetes_groups` configuration and, if not, denies the user access.
+
+##### Automatically
 
-Teleport uses [Kubernetes impersonation](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#user-impersonation)
-to map OIDC and SAML users to Kubernetes users and groups.
+If the user has not explicitly determined a Kubernetes user and Kubernetes
```

Comment:
Mention that:
- Only one kubernetes_user is allowed to be impersonated per request. If the role allows multiple users, you must select one using `--as` otherwise the request is denied.
- When the role/user defines no `kubernetes_users`, Teleport will use the Teleport username as the user to impersonate.